syncserver/src/tokenserver/handlers.rs

```diff
@@ -250,7 +276,10 @@ pub async fn heartbeat(DbWrapper(db): DbWrapper) -> Result<HttpResponse, Error>
         Value::String(env!("CARGO_PKG_VERSION").to_owned()),
     );
 
-    match db.check().await {
+    match apply_timeout(db.timeout(), db.check())
+        .await
+        .map_err(|_| TokenserverError::elapsed())
+    {
```

Comment:
Do we need the `map_err` now that `apply_timeout` does it?
```suggestion
    match apply_timeout(db.timeout(), db.check())
        .await
    {
```